### DISCUSSION TITLE
ReactNative Controller onChange saves event object instead of text

### DISCUSSION THREAD

**Discussion by pke** [react-hook-form/react-hook-form#2985] [General] [answered]
In my ReactNative using `useForm@6.8.4` and this Controller code:

```tsx
const { control, handleSubmit, setError } = useForm<Contact, ApiError>()
<Controller
  name="firstName"
  control={control}
  as={<TextInput/>}
  defaultValue=""
/>

const onSend(contact: Contact) {
  // contact contains the an event object for each field instead of the text values
  // see below
}
<Button onPress={handleSubmit(onSend)}/>
```

`TextInput` is the default RN TextInput. 
It works when using `getValues`

This is what `contact` contains:
```
"firstName": {"_dispatchInstances": null, "_dispatchListeners": null, "_targetInst": null, "bubbles": null, "cancelable": null, "currentTarget": [Function currentTarget], "defaultPrevented": null, "dispatchConfig": null, "eventPhase": null, "isDefaultPrevented": [Function functionThatReturnsFalse], "isPropagationStopped": [Function functionThatReturnsFalse], "isTrusted": null, "nativeEvent": null, "target": null, "timeStamp": [Function timeStamp], "type": null}, "lastName": "", "phoneNumber": "", "title": {"_dispatchInstances": null, "_dispatchListeners": null, "_targetInst": null, "bubbles": null, "cancelable": null, "currentTarget": [Function currentTarget], "defaultPrevented": null, "dispatchConfig": null, "eventPhase": null, "isDefaultPrevented": [Function functionThatReturnsFalse], "isPropagationStopped": [Function functionThatReturnsFalse], "isTrusted": null, "nativeEvent": null, "target": null, "timeStamp": [Function timeStamp], "type": null}}
```

Also there is a warning displayed saying "this synthetic event is reused for perfomance reasons..."
![Simulator Screen Shot - iPhone 11 - 2020-09-23 at 09 16 53](https://user-images.githubusercontent.com/164625/93979427-9cb3f480-fd7d-11ea-8ceb-599c2558dbd0.png)

While the stack of the warning does not reveal anything about `useForm` it is related to the saved data in the values returned by `getValues`.

**Comment by bluebill1049** [comment, marked as answer]
hey @pke use `render` instead.

https://react-hook-form.com/api#Controller

**Reply by pke** [reply]
@bluebill1049 updated the question. It happens when using `handleSubmit`. Maybe I am using it wrong (again), but I remember there was a bug with it in earlier versions of useForm.

```jsx
const onNext = (contact: Contact) => {
}
<Button onPress={handleSubmit(onNext)}/>
```

Something wrong with this code?

**Reply by bluebill1049** [reply]
nothing wrong with the code above.

**Reply by pke** [reply]
Ah ok, the problem really comes from using `as` since then the form tries to attach to the "change" event and RN decided to use `onTextChange` as the actual event that is fired with the actual value of the text.
